src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { InitialStates, Todo } from './types/Todo';
+import { getTodos, getUser } from './api';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [toggleModal, setToggleModal] = useState<boolean>(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>();
+  const [user, setUser] = useState<User>();
+  const [filterTodos, setFilterTodos] = useState<InitialStates>({
+    query: '',
+    select: 'all',
+  });
+
+  const filterResponse = useCallback(
+    (responseTodos: Todo[]) => {
+      let filteredTodos = [...responseTodos];
+
+      if (filterTodos.query !== '') {
+        filteredTodos = responseTodos.filter(todo =>
+          todo.title.toLowerCase().includes(filterTodos.query.toLowerCase()),
+        );
+        setTodos(filteredTodos);
+      }
+
+      if (filterTodos.select === 'completed') {
+        filteredTodos = filteredTodos.filter(todo => todo.completed === true);
+        setTodos(filteredTodos);
+      } else if (filterTodos.select === 'active') {
+        filteredTodos = filteredTodos.filter(todo => todo.completed === false);
+        setTodos(filteredTodos);
+      } else {
+        setTodos(filteredTodos);
+      }
+    },
+    [filterTodos.query, filterTodos.select],
+  );
```

Comment:
It makes sense to move this method away from the component

IMO the best would be creating sort of the hook `useTodos` by combining this and the effect hook, you have below